File: Player/src/main/java/ru/testsimpleapps/coloraudioplayer/ui/animation/BaseAnimationListener.java
package ru.testsimpleapps.coloraudioplayer.ui.animation;

import android.util.Log;
import android.view.View;
import android.view.animation.Animation;

public abstract class BaseAnimationListener implements Animation.AnimationListener {

    public static final String TAG = BaseAnimationListener.class.getSimpleName();

    protected View mView;
    protected boolean mIsAnimating = false;

    public BaseAnimationListener() {
    }

    public BaseAnimationListener(final View view) {
        mView = view;
    }

    @Override
    public void onAnimationStart(Animation animation) {
        Log.d(TAG, "onAnimationStart()");
        mIsAnimating = true;
    }

    @Override
    public void onAnimationEnd(Animation animation) {
        Log.d(TAG, "onAnimationEnd()");
        mIsAnimating = false;
    }

    @Override
    public void onAnimationRepeat(Animation animation) {
        Log.d(TAG, "onAnimationRepeat()");
    }

    public boolean isAnimating() {
        return mIsAnimating;
    }

}

File: player/src/main/java/ru/testsimpleapps/coloraudioplayer/ui/animation/BaseAnimationListener.kt
package ru.testsimpleapps.coloraudioplayer.ui.animation

import android.util.Log
import android.view.View
import android.view.animation.Animation

abstract class BaseAnimationListener : Animation.AnimationListener {

    protected lateinit var mView: View
    var isAnimating = false
        protected set

    constructor() {}

    constructor(view: View) {
        mView = view
    }

    override fun onAnimationStart(animation: Animation) {
        Log.d(TAG, "onAnimationStart()")
        isAnimating = true
    }

    override fun onAnimationEnd(animation: Animation) {
        Log.d(TAG, "onAnimationEnd()")
        isAnimating = false
    }

    override fun onAnimationRepeat(animation: Animation) {
        Log.d(TAG, "onAnimationRepeat()")
    }

    companion object {

        val TAG = BaseAnimationListener::class.java.simpleName
    }

}
